### PR TITLE
#9228 - Tabular file not generated with only Output:Table:Annual

### LINKED ISSUE #9228: Tabular file not generated with only Output:Table:Annual
Issue overview
--------------
The tabular output file (usually <filename>Table.html) does not get generated if only Output:Table:Annual input objects are present. If Output:Table:SummaryReports or Output:Table:Monthly or Output:Table:TimeBins are also present, the tabular output file is generated and the Output:Table:Annual are shown but by themselves, the Output:Table:Annual do not generate the tabular output file.

To reproduce the problem, use PurchAirTables_wAnnual.idf from the ExampleFiles and remove or comment out Output:Table:SummaryReports, Output:Table:Monthly, and Output:Table:TimeBins objects but leave Output:Table:Annual objects.

### Details
Some additional details for this issue (if relevant):
 - Tested on Windows
 - 9.6.0
 - https://unmethours.com/question/66734/e-outputtableannual-without-summary-report/
 
### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added : https://github.com/NREL/EnergyPlusDevSupport/blob/25f8c6836deb770d86bbd042e2c10a193b424fd3/DefectFiles/9000s/9228/PurchAirTables_wAnnual.idf
 - [x] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------
 - Fixes #9228
 - Tabular file not generated with only Output:Table:Annual

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [x] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [x] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [x] If structural output changes, add to output rules file and add OutputChange label
 - [x] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/OutputReportTabularAnnual.cc [hunk 1]
```diff
@@ -107,6 +107,9 @@ void GetInputTabularAnnual(EnergyPlusData &state)
 
     objCount = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, currentModuleObject);
     if (objCount > 0) {
+
+        state.dataOutRptTab->WriteTabularFiles = true;
```
Comment (jmarrec) [inline src/EnergyPlus/OutputReportTabularAnnual.cc:111]: actual fix

File: src/EnergyPlus/SimulationManager.cc [hunk 2]
```diff
@@ -1653,6 +1653,7 @@ namespace SimulationManager {
             ReportingRequested = (state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, "Output:Table:SummaryReports") > 0 ||
                                   state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, "Output:Table:TimeBins") > 0 ||
                                   state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, "Output:Table:Monthly") > 0 ||
+                                  state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, "Output:Table:Annual") > 0 ||
```
Comment (jmarrec) [inline src/EnergyPlus/SimulationManager.cc:1656]: Don't warn about "no output requested" if you requested an Output:Table:Annual

File: tst/EnergyPlus/unit/OutputReportTabularAnnual.unit.cc [hunk 3]
```diff
@@ -85,7 +85,9 @@ TEST_F(EnergyPlusFixture, OutputReportTabularAnnual_GetInput)
 
     state->dataGlobal->DoWeathSim = true;
 
+    EXPECT_FALSE(state->dataOutRptTab->WriteTabularFiles);
     GetInputTabularAnnual(*state);
+    EXPECT_TRUE(state->dataOutRptTab->WriteTabularFiles);
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/OutputReportTabularAnnual.unit.cc:90]: Minimal test.

**Review by jmarrec — COMMENTED** [review]
Checked the defect file on dev support before (didn't have tables), and confirmed it has eplustbl.htm after.

**Review by mjwitte — APPROVED** [review]
Nice fix, works as advertised. I've merged in develop, will wait for one CI result or so then merge this.

**Pull request merged**